### ISSUE TITLE
grating coupling - cannot verify with COMSOL

### ISSUE THREAD

**Issue by VovaZen** [stanfordnqp/spins-b#11]
I play with the grating coupling, and produced grating I then check in COMSOL with FEM (Boundary Mode Analysis + Frequency domain). I optimize grating for single wavelength, and the script says the optimized efficiency (mon_power) is 0.32, but in COMSOL it calculates it as only 8% or lower. The field plots from COMSOL and script looks very much the same. However, I found that if I increase the refractive index of silicon by 0.0125, then the coupling efficiency calculated in COMSOL is nearly the same as predicted by the script. Even if in the script I directly write the refractive index of Si as 3.48, the optimum in COMSOL is still for n = 3.605. Strange :)

**Comment by ludi1001** [comment]
Have you run a convergence test? Perhaps dx is too large.

Another possibility is to double check that the grating is actually the same. We have personal experiences where the grating edges near the waveguide were wrong, so we were simulating two different gratings (in FDFD vs FDTD).

**Comment by ludi1001** [comment]
Can you verify whether the distance between the grating and the waveguide is actually correct? The waveguide actually might not start at zero...

**Comment by VovaZen** [comment]
Would like to verify the distance, but how? I take grating positions using the script you provided, and I know there the waveguide edge is added manually as 0... As for convergence test, do you mean to run in COMSOL? Then it is automatically there and I trust its results to 99%
Regarding to large dx - do you mean in spins? Cause in COMSOL I use defferent mesh with mesh size lambda_0/n_refr/10 as recommended by COMSOL :)

**Comment by VovaZen** [comment]
You was right, the waveguide doesn't start at 0. What I found is that the first grating element always begins at 2*minimum_feature_size, and solid waveguide is allowed to end between 0 and 1*minimum_feature_size. I found exact position of the waveguide end using pkl files, monitor for epsilon (permittivity), take cross-section through the waveguide and then by taking reported average permittivity I recover the length of the waveguide which sticks out beyond 0. By the way, why not to save waveguide coordinates directly into parametrization?
This bug is linked to bugs #15 and #14
